servers/rendering/renderer_rd/environment/sky.cpp

```diff
@@ -1323,7 +1323,7 @@ void SkyRD::update_radiance_buffers(Ref<RenderSceneBuffersRD> p_render_buffers,
 
 		// Note, we ignore environment_get_sky_orientation here as this is applied when we do our lookup in our scene shader.
 
-		if (shader_data->uses_quarter_res) {
+		if (shader_data->uses_quarter_res && roughness_layers >= 3) {
```

Comment:
We need to do more testing with this. The most likely scenario where we're using quarter or half res is on mobile, and on mobile by default reflection texture atlas is turned off so it is likely we won't have working quarter or half res sky in the use case its most needed.

I think we need to consider creating separate textures for quarter and half res, seeing the most likely use case we won't be creating the reflection textures there isn't that much of a risk of allocated double memory.

Reply:
After discussing with Clay how this works, it is a little counter intuitive because we're using the same buffers, but they are being created for sky even if the setting in project settings is turned off. The added problem here is the roughness layer count now comes into play.